File: NoteiOS-master/Objective-C/00BM需求/签名 画图/Signature/Signature/ViewController.m
//
//  ViewController.m
//  Signature
//
//  Created by ___liangdahong on 16/8/31.
//  Copyright © 2016年 https://github.com/asiosldh/. All rights reserved.
//

#import "ViewController.h"
#import "PJRSignatureView.h"
#import "BMAlertSignView.h"

@interface ViewController ()

@property (nonatomic, strong) PJRSignatureView *signatureView;

@end

@implementation ViewController

- (PJRSignatureView *)signatureView {
    if (!_signatureView) {
        CGFloat w = [UIScreen mainScreen].bounds.size.width;
        CGFloat h = [UIScreen mainScreen].bounds.size.height;
        _signatureView = [PJRSignatureView signatureViewWithFrame:CGRectMake(0, 0, w, h-40-64) promptView:nil];
        _signatureView.lineWidth = 10;
        _signatureView.lineColor = [UIColor orangeColor];
    }
    return _signatureView;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self.view addSubview:self.signatureView];

}

- (IBAction)clearClick:(id)sender {
    [self.signatureView clear];
}

- (IBAction)okClick:(id)sender {
    [BMAlertSignView showWithImage:self.signatureView.signatureImage];
}

- (IBAction)add:(id)sender {
    UIButton *button = [UIButton buttonWithType:UIButtonTypeContactAdd];
    button.frame = CGRectMake(100, 100, 20, 20);
    self.signatureView.promptView = button;
}

@end

File: NoteiOS-master/Objective-C/Objective-C与Swift对比/block和闭包之对象绑定/SwiftClosure/SwiftClosure/ViewController.swift
//
//  ViewController.swift
//  SwiftClosure
//
//  Created by ___liangdahong on 16/9/9.
//  Copyright © 2016年 https://github.com/asiosldh/. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        let btn = UIButton.init(frame: CGRectMake(100, 100, 100, 100))
        btn.backgroundColor = UIColor.orangeColor()
        self.view.addSubview(btn)
        btn.ih_addTouchUpInsideClosure { 
            print("1")
        }
        btn.ih_addTouchUpInsideClosure {
            print("2")
        }
    }
}



